crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -192,37 +193,55 @@ impl C8YHttpProxyActor {
         &mut self,
         device_id: String,
     ) -> Result<String, C8YRestError> {
-        let url_get_id: String = self.end_point.get_url_for_internal_id(device_id);
+        let url_get_id: String = self.end_point.get_url_for_internal_id(&device_id);
         if self.end_point.token.is_none() {
             self.get_fresh_token().await?;
         }
-        let request = HttpRequestBuilder::get(&url_get_id)
-            .bearer_auth(self.end_point.token.clone().unwrap_or_default())
-            .build()?;
 
-        let res = match self.peers.http.await_response(request).await? {
-            Ok(response) => match response.status() {
-                StatusCode::OK => Ok(Ok(response)),
-                StatusCode::UNAUTHORIZED | StatusCode::FORBIDDEN => {
-                    self.get_fresh_token().await?;
-                    let request = HttpRequestBuilder::get(&url_get_id)
-                        .bearer_auth(self.end_point.token.clone().unwrap_or_default())
-                        .build()?;
-                    Ok(self.peers.http.await_response(request).await?)
+        let attempt = 1;
```

Comment:
Where do you increase this `attempt`...?

Reply:
Good catch ;)